scripts/azureml-assets/azureml/assets/config.py

```diff
@@ -345,7 +351,73 @@ def __init__(self, branch: str, uri: str):
         self._branch = branch
         super().__init__(PathType.GIT, uri)
 
+class DataConfig(Config):
+    """Data Config class."""
+
+    """
+    Example:
+
+    path: #Could be azure storage, git , hugging face
+
+        ## Azure Blobstore example
+        type: azureblob
+        storage_name: my_storage
+        container_name: my_container
+        container_path: foo/bar
+    """
+    def __init__(self, file_name: Path):
+        """Initialize object for the Data Properties extracted from extra_config data.yaml."""
+        super().__init__(file_name)
+        self._path = None
+        self._validate()
+
+    def _validate(self):
+        """Validate the yaml file."""
+        Config._validate_exists('data.path', self.path)
+        Config._validate_enum('data.path.type', self.path.type.value, PathType, True)
+        Config._validate_enum('data.type', self._type, DataType, True)
+
+    @property
+    def path(self) -> AssetPath:
+        """Data Path."""
+        if self._path:
+            return self._path
+        path = self._yaml.get('path', {})
+        if path and path.get('type'):
+            path_type = path.get('type')
+            if path_type == PathType.AZUREBLOB.value:
+                self._path = AzureBlobstoreAssetPath(
+                    storage_name=path['storage_name'],
+                    container_name=path['container_name'],
+                    container_path=path['container_path'],
+                )
+            elif path_type == PathType.GIT.value:
+                self._path = GitAssetPath(branch=path['branch'], uri=path['uri'])
+            elif path_type == PathType.LOCAL.value:
+                self._path = LocalAssetPath(local_path=path['uri'])
+            elif path_type == PathType.HTTP.value or path_type == PathType.FTP.value:
+                raise NotImplementedError("Support for HTTP and FTP is being added.")
+        else:
+            raise Exception("path parameters are invalid")
+        return self._path
+
+    @property
+    def _publish(self) -> Dict[str, object]:
+        """Data publish properties."""
+        return self._yaml.get('publish')

```

Comment:
I think you can delete this. Looks like it was pasted in from another class.

Reply:
Under publish key, I am trying to maintain the type of data assets - uri_file/uri_folder/mtlable  as it was for model .
So thats why kept it here.

ref - https://msdata.visualstudio.com/Vienna/_git/azureml-asset/pullrequest/1004391?_a=files&path=/assets/data/multi_label_classification/data.yaml
Let me know if there is any drawback .
Thanks